src/libexpr-tests/yaml.cc

```
@@ -0,0 +1,425 @@
+            if (remSize >= 3 && (filler.find(data + i, 0, 3) != eop.find(data + i, 0, 3))) {
+                seqSize = 3;
+            } else if (remSize >= 3 && space.find(data + i, 0, 3) != space.npos) {
+                replaceWith = ' ';
+                seqSize = 3;
+            } else if (remSize >= 3 && newLine.find(data + i, 0, 3) != newLine.npos) {
+                seqSize = 3;
+            } else if (remSize >= 2 && tab.find(data + i, 0, 2) != tab.npos) {
+                replaceWith = '\t';
+                seqSize = 2;
+            } else {
+                continue;
+            }
+            ret.append(str, last, i - last);
+            if (replaceWith != '\0') {
+                ret.append(&replaceWith, 1);
+            }
+            last = i + seqSize;
+            i += seqSize - 1;
+        }
+    }
+    ret.append(str.begin() + last, str.size() - last);
+    return ret;
+}
+
+bool parseJSON(EvalState & state, std::istream & s_, Value & v)
+{
+    auto parser = makeJSONSaxParser(state, v);
+    return nlohmann::json::sax_parse(s_, parser.get(), nlohmann::json::input_format_t::json, false);
+}
+
+Value parseJSONStream(EvalState & state, std::string_view json, std::function<FromYAMLFun> fromYAML)
+{
+    std::stringstream ss;
+    ss << json;
+    std::list<Value> list;
+    Value root, refJson;
+    std::streampos start = 0;
+    try {
+        while (ss.peek() != EOF && json.size() - ss.tellg() > 1) {
+            parseJSON(state, ss, refJson);
+            list.emplace_back(refJson);
+            // sanity check: builtins.fromJSON and builtins.fromYAML should return the same result when applied to a
+            // JSON string
+            root.mkString(std::string_view(json.begin() + start, ss.tellg() - start));
+            Value rymlJson = fromYAML(state, root, {});
+            EXPECT_EQ(printValue(state, refJson), printValue(state, rymlJson));
+            start = ss.tellg() + std::streampos(1);
+        }
```

sugg: maybe a comment why this empty catch here